### ISSUE TITLE
Issue when creating new import

### ISSUE THREAD

**Issue by KaisAhmad** [matomo-org/plugin-GoogleAnalyticsImporter#307]
I used to import many different websites from Google Analytics using this plugin. Since the new update (4.4.1), all imports no longer work. I then get this error:

```
ERROR [2022-12-06 07:42:51] 5033  Uncaught exception: /var/www/html/core/Date.php(1144): Date format must be: YYYY-MM-DD, or 'today' or 'yesterday' or any keyword supported by the strtotime function (see http://php.net/strtotime for more information):  [Query: , CLI mode: 1]
DEBUG [2022-12-06 07:42:51] 5033  Loaded plugins: CoreVue, CorePluginsAdmin, CoreAdminHome, CoreHome, WebsiteMeasurable, IntranetMeasurable, Diagnostics, CoreVisualizations, Proxy, API, Widgetize, Transitions, LanguagesManager, Actions, Dashboard, MultiSites, Referrers, UserLanguage, DevicesDetection, Goals, Ecommerce, SEO, Events, UserCountry, GeoIp2, VisitsSummary, VisitFrequency, VisitTime, VisitorInterest, RssWidget, Feedback, Monolog, Login, TwoFactorAuth, UsersManager, SitesManager, Installation, CoreUpdater, CoreConsole, ScheduledReports, UserCountryMap, Live, PrivacyManager, ImageGraph, Annotations, MobileMessaging, Overlay, SegmentEditor, Insights, Morpheus, Contents, BulkTracking, Resolution, DevicePlugins, Heartbeat, Intl, Marketplace, ProfessionalServices, UserId, CustomJsTracker, Tour, PagePerformance, CustomDimensions, DBStats, Bandwidth, GoogleAnalyticsImporter, IPtoCompany
                                                                                                                                                                       
[Exception]                                                                                                                                                          
Date format must be: YYYY-MM-DD, or 'today' or 'yesterday' or any keyword supported by the strtotime function (see http://php.net/strtotime for more information):   
                                                                                                                                                                       
Exception trace:
 () at /var/www/html/core/Date.php:1144
 Piwik\Date::getInvalidDateFormatException() at /var/www/html/core/Date.php:153
 Piwik\Date::factory() at /var/www/html/plugins/GoogleAnalyticsImporter/Commands/ImportReports.php:442
 Piwik\Plugins\GoogleAnalyticsImporter\Commands\ImportReports->checkIfCanProcess() at /var/www/html/plugins/GoogleAnalyticsImporter/Commands/ImportReports.php:86
 Piwik\Plugins\GoogleAnalyticsImporter\Commands\ImportReports->executeImpl() at /var/www/html/plugins/GoogleAnalyticsImporter/Commands/ImportReports.php:60
 Piwik\Plugins\GoogleAnalyticsImporter\Commands\ImportReports->execute() at /var/www/html/vendor/symfony/console/Symfony/Component/Console/Command/Command.php:257
 Symfony\Component\Console\Command\Command->run() at /var/www/html/vendor/symfony/console/Symfony/Component/Console/Application.php:874
 Symfony\Component\Console\Application->doRunCommand() at /var/www/html/vendor/symfony/console/Symfony/Component/Console/Application.php:195
 Symfony\Component\Console\Application->doRun() at n/a:n/a
 call_user_func() at /var/www/html/core/Console.php:135
 Piwik\Console->Piwik\{closure}() at /var/www/html/core/Access.php:670
 Piwik\Access::doAsSuperUser() at /var/www/html/core/Console.php:136
 Piwik\Console->doRunImpl() at /var/www/html/core/Console.php:87
 Piwik\Console->doRun() at /var/www/html/vendor/symfony/console/Symfony/Component/Console/Application.php:126
 Symfony\Component\Console\Application->run() at /var/www/html/console:32


googleanalyticsimporter:import-reports [--property="..."] [--account="..."] [--view="..."] [--dates="..."] [--idsite="..."] [--cvar-count="..."] [--skip-archiving] [--mobile-app] [--timezone="..."] [--extra-custom-dimension="..."]
```

Even if I do not enter a date at the creation date and end date I keep getting this error. Could someone help me with this?

I have tried to manually put it into the database hoping to fix it but still same issue.

SQL:
`GoogleAnalyticsImporter. importStatus_14 | {"status":"ongoing","idSite":14,"ga":{"import_type":"Universal Analytics","property":"<HIDDEN>","account":"<HIDDEN>","view": "<HIDDEN>"},"last_date_imported":"2022-11-10","main_import_progress":"2022-11-10","import_start_time":1670252638,"import_end_time": null, "last_job_start_time":1670315481, "last_day_archived": "2022-11-10", "import_range_start":null, "import_range_end":null, "extra_custom_dimensions":[], "days_finished_since_rate_limit":0, "reimport_ranges":[], "isGA4":false, "is_verbose_logging_enabled":true}`

(the weird part is that it states that import 14 is ongoing but the dashboard shows it as killed).

**Comment by AltamashShaikh** [comment]
@KaisAhmad Can you share the output of this query ?
```
Select * FROM `matomo_option` WHERE option_name='GoogleAnalyticsImporter_nextAvailableAt'
```

**Comment by AltamashShaikh** [comment]
@KaisAhmad if its empty can you change this line of code in your file and rerun the import? `plugins/GoogleAnalyticsImporter/Commands/ImportReports.php`
https://github.com/matomo-org/plugin-GoogleAnalyticsImporter/blob/4.x-dev/Commands/ImportReports.php#L434 to
```
if (!$nextAvailableAt) {
```

**Comment by KaisAhmad** [comment]
@AltamashShaikh

Output of the query:

+-----------------------------------------+--------------+----------+
| option_name                             | option_value | autoload |
+-----------------------------------------+--------------+----------+
| GoogleAnalyticsImporter_nextAvailableAt | 1670371200   |        0 |
+-----------------------------------------+--------------+----------+

**Comment by AltamashShaikh** [comment]
@KaisAhmad Thanks, got the issue its being treated as a string and hence the error

**Comment by AltamashShaikh** [comment]
@KaisAhmad Created #308 to address the issue, can you apply [this](https://github.com/matomo-org/plugin-GoogleAnalyticsImporter/pull/308/files) change and check if its working ?

**Comment by KaisAhmad** [comment]
@AltamashShaikh Original error is gone. The moment I press resume I get back that it doesn't start automatically until the next day.

```
The import will be restarted automatically at 2022-12-07 12:00 am
ERROR [2022-12-06 09:59:06] 107  Uncaught exception: /var/www/html/plugins/GoogleAnalyticsImporter/Commands/ImportReports.php(90): The import will be restarted automatically at 2022-12-07 12:00 am [Query: , CLI mode: 1]
DEBUG [2022-12-06 09:59:06] 107  Loaded plugins: CoreVue, CorePluginsAdmin, CoreAdminHome, CoreHome, WebsiteMeasurable, IntranetMeasurable, Diagnostics, CoreVisualizations, Proxy, API, Widgetize, Transitions, LanguagesManager, Actions, Dashboard, MultiSites, Referrers, UserLanguage, DevicesDetection, Goals, Ecommerce, SEO, Events, UserCountry, GeoIp2, VisitsSummary, VisitFrequency, VisitTime, VisitorInterest, RssWidget, Feedback, Monolog, Login, TwoFactorAuth, UsersManager, SitesManager, Installation, CoreUpdater, CoreConsole, ScheduledReports, UserCountryMap, Live, PrivacyManager, ImageGraph, Annotations, MobileMessaging, Overlay, SegmentEditor, Insights, Morpheus, Contents, BulkTracking, Resolution, DevicePlugins, Heartbeat, Intl, Marketplace, ProfessionalServices, UserId, CustomJsTracker, Tour, PagePerformance, CustomDimensions, DBStats, Bandwidth, GoogleAnalyticsImporter, IPtoCompany


                                                                     
  [Exception]                                                        
  The import will be restarted automatically at 2022-12-07 12:00 am  
                                                                     


Exception trace:
 () at /var/www/html/plugins/GoogleAnalyticsImporter/Commands/ImportReports.php:90
 Piwik\Plugins\GoogleAnalyticsImporter\Commands\ImportReports->executeImpl() at /var/www/html/plugins/GoogleAnalyticsImporter/Commands/ImportReports.php:60
 Piwik\Plugins\GoogleAnalyticsImporter\Commands\ImportReports->execute() at /var/www/html/vendor/symfony/console/Symfony/Component/Console/Command/Command.php:257
 Symfony\Component\Console\Command\Command->run() at /var/www/html/vendor/symfony/console/Symfony/Component/Console/Application.php:874
 Symfony\Component\Console\Application->doRunCommand() at /var/www/html/vendor/symfony/console/Symfony/Component/Console/Application.php:195
 Symfony\Component\Console\Application->doRun() at n/a:n/a
 call_user_func() at /var/www/html/core/Console.php:135
 Piwik\Console->Piwik\{closure}() at /var/www/html/core/Access.php:670
 Piwik\Access::doAsSuperUser() at /var/www/html/core/Console.php:136
 Piwik\Console->doRunImpl() at /var/www/html/core/Console.php:87
 Piwik\Console->doRun() at /var/www/html/vendor/symfony/console/Symfony/Component/Console/Application.php:126
 Symfony\Component\Console\Application->run() at /var/www/html/console:32


googleanalyticsimporter:import-reports [--property="..."] [--account="..."] [--view="..."] [--dates="..."] [--idsite="..."] [--cvar-count="..."] [--skip-archiving] [--mobile-app] [--timezone="..."] [--extra-custom-dimension="..."]
```

**Comment by KaisAhmad** [comment]
Just forced the update by changing the `GoogleAnalyticsImporter_nextAvailableAt` and everyting works fine again! The issue can be closed. Thanks for the help 👍🏽

**Comment by snake14** [comment]
@KaisAhmad I'm happy to hear that things are working correctly. I'll go ahead and close the issue, but please let us know if we need to re-open it for any reason.

**Comment by snake14** [comment]
@KaisAhmad I just released version 4.4.2 of the plugin containing the fix.

**Comment by KaisAhmad** [comment]
@snake14 alright good work. Thanks!
